make-util.js

```diff
@@ -144,16 +148,50 @@ var getPackFileName = function (packageJsonPath) {
 }
 exports.getPackFileName = getPackFileName
 
-var buildNodeTask = function (taskPath, outDir) {
+var buildNodeTask = function (taskPath, outDir, skipNpm, syncBundleDeps) {
     var originalDir = pwd();
     cd(taskPath);
     var packageJsonPath = rp('package.json');
-    if (test('-f', packageJsonPath)) {
-        var packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
-        if (packageJson.devDependencies && Object.keys(packageJson.devDependencies).length != 0) {
+    if (!skipNpm && test('-f', packageJsonPath)) {
+        // verify no dev dependencies
+        var package = JSON.parse(fs.readFileSync(packageJsonPath).toString());
+        if (package.devDependencies && Object.keys(package.devDependencies).length != 0) {
             fail('The package.json should not contain dev dependencies. Move the dev dependencies into a package.json file under the Tests sub-folder. Offending package.json: ' + packageJsonPath);
         }
 
+        if (syncBundleDeps) {
```

Comment:
this section generates the bundle section for common modules. so consumers dont have to think about it, the file is always generated and correct.